### DISCUSSION TITLE
NodeJS - Can we run widgets and more calculation on the server side, and cache in server side?

### DISCUSSION THREAD

**Discussion by linonetwo** [TiddlyWiki/TiddlyWiki5#5402] [Q&A]
Notion and roam research and logseq calculate their backlinks in the server side. And relink also happened in the server side.

Running and caching things at the server-side have 2 benefits:

## Reduce the memory usage, allow huge KB to mount

Currently, if we have 1GB of toddler files in the disk, we will load 1GB in the nodejs, and 1GB in the browser, this is a waste.

And some widget like backlink, that relies on loading all the tiddlers, can work on the server-side, and send the result to the client.

##  Run heavy things in server-side

Currently, if we want to refactor our KB, we can use relink plugin, but relink progress will run in the browser, relies on we fully load all tiddlers. So if we lazy load the tiddlers, it will not work.
Run these tasks on the server-side will solve this problem.

**Comment by Jermolene** [comment]
Hi @linonetwo in #3461 we've been experimenting with allowing the browser to trigger commands to be executed on the server (amongst other things). I'll add a note there.

**Comment by hoelzro** [comment]
This reminds me of #3000 as well!

**Reply by linonetwo** [reply]
This will be important, I'm going to use TW as storage of web clipper https://github.com/webclipper/web-clipper/issues/688 , this will resulted in GBs of tiddlers, so I hope searching and indexing can be done in the backend.
